src/jit/morph.cpp

```diff
@@ -2800,7 +2800,11 @@ GenTreeCall* Compiler::fgMorphArgs(GenTreeCall* callNode)
         assert(argx->gtType != TYP_STRUCT);
 
         /* Increment the argument register count and argument index */
-        if (!varTypeIsFloating(argx->gtType))
+        bool fltPassedInInt = false;
+#ifdef ARM_SOFTFP
+        fltPassedInInt = true;
+#endif
+        if (!varTypeIsFloating(argx->gtType) || fltPassedInInt)
```

Comment:
@BruceForstall so, once more about this particular line. Isn't that needed? Don't we need to intArgRegNum++ in such a case, and why we don't do that for Varargs?